federation/cmd/federated-apiserver/app/server.go

```diff
@@ -390,9 +240,8 @@ func Run(s *options.APIServer) error {
 
 	var serviceAccountGetter serviceaccount.ServiceAccountTokenGetter
 	if s.ServiceAccountLookup {
-		// If we need to look up service accounts and tokens,
-		// go directly to etcd to avoid recursive auth insanity
-		serviceAccountGetter = serviceaccountcontroller.NewGetterFromStorageInterface(etcdStorage)
+		// TODO: Maybe do not expose this option in genericapiserver, if most servers do not need it?
```

Comment:
Yes to clarify, s.ServiceAccountLookup dictates if we should do the lookup. We have serviceAccountGetter that actually does the lookup and as @liggitt pointed out, there is a client implementation as well, which we can use.

Also there are 3 issues here:
* Should serviceAccountLookup be set to true for ubernetes? I think we all agree on "no". Ubernetes server does not serve tokens and service accounts.
* Should serviceAccountLookup be set to true for core kubernetes server? Seems like we want to set it to true, but we do not currently. I can file a separate issue for that and fix it. I dont know enough about it to understand all the implications of that change.
* Should genericapiserver provide the ability? I guess if we decide to set it to true in core kubernetes server, then we can let that option be in genericapiserver. My proposal to delete it was based on the current state, where we do not set it anywhere.

Also @liggitt and @deads2k Do you guys set is to true in openshift server?